cpp/ql/lib/semmle/code/cpp/Type.qll

```diff
@@ -1666,6 +1666,28 @@ class RoutineType extends Type, @routinetype {
   }
 }
 
+abstract private class TemplateParameterImpl extends Locatable {
+  override string getAPrimaryQlClass() { result = "TemplateParameterImpl" }
+}
+
+/**
+ * A C++ template parameter.
+ *
+ * In the example below, `T` and `I` are template parameters:
+ * ```
+ * template <class T, int I>
+ * class C { };
```

Comment:
Maybe it would be nice to add a template template parameter as well:
```suggestion
 * In the example below, `T`, `I`, and `V` are template parameters:
 * ```
 * template <class T, int I, template<typename> class V>
 * class C { };
```

Reply:
Done in https://github.com/github/codeql/pull/18308/commits/ccd3681f833e00c2b1ef1e70a8a1c69936045851